packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -595,10 +603,15 @@ function shallowCompatibilityTest(
 		return mapOrArray ? CompatibilityLevel.Normal : CompatibilityLevel.None;
 	}
 
-	if (mapOrArray) {
+	if (schema.kind === NodeKind.Array) {
```

Comment:
While you're fixing the typo in this file, could you also move the comment below - the one that says "Assume record-like object" - up here above this line (or wherever appropriate)? I think that will help readers know what is going on here with the array and map schema check.

Reply:
Fixed. Good catch: its way clearer now.